presto/dataset.py

```diff
@@ -141,13 +147,111 @@ def __getitem__(self, idx):
         row = self.df.iloc[idx, :]
         eo, mask_per_token, latlon, month, target = self.row_to_arrays(row)
         mask_per_variable = np.repeat(mask_per_token, BAND_EXPANSION, axis=1)
-        num_masked_tokens = sum(sum(mask_per_token))
         return (
             self.normalize_and_mask(eo),
             target,
             np.ones(self.NUM_TIMESTEPS) * (DynamicWorld2020_2021.class_amount),
             latlon,
             month,
-            num_masked_tokens,
             mask_per_variable,
         )
+
+
+class WorldCerealInferenceDataset(Dataset):
+    _NODATAVALUE = 65535
+    Y = "worldcereal_cropland"
+    BAND_MAPPING = {
+        "B02": "B2",
+        "B03": "B3",
+        "B04": "B4",
+        "B05": "B5",
+        "B06": "B6",
+        "B07": "B7",
+        "B08": "B8",
+        # B8A is missing
+        "B11": "B11",
+        "B12": "B12",
+        "VH": "VH",
+        "VV": "VV",
+        "precipitation-flux": "total_precipitation",
+        "temperature-mean": "temperature_2m",
+    }
+
+    def __init__(self):
+        self.path_to_files = data_dir / "inference_areas"
+        self.all_files = list(self.path_to_files.glob("*.nc"))
+
+    def __len__(self):
+        return len(self.all_files)
+
+    @classmethod
+    def nc_to_arrays(
+        cls, filepath: Path
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        ds = cast(xr.Dataset, rioxarray.open_rasterio(filepath, decode_times=False))
+        epsg_coords = ds.rio.crs.to_epsg()
+
+        num_instances = len(ds.x) * len(ds.y)
+        num_timesteps = len(ds.t)
+        eo_data = np.zeros((num_instances, num_timesteps, len(BANDS)))
+        mask = np.zeros((num_instances, num_timesteps, len(BANDS_GROUPS_IDX)))
+        # for now, B8A is missing
+        mask[:, :, IDX_TO_BAND_GROUPS["B8A"]] = 1
+
+        for org_band, presto_val in cls.BAND_MAPPING.items():
+            # flatten the values
+            values = np.swapaxes(ds[org_band].values.reshape((num_timesteps, -1)), 0, 1)
+            idx_valid = values != cls._NODATAVALUE
+
+            if presto_val in ["VV", "VH"]:
+                # convert to dB
+                values = 20 * np.log10(values) - 83
+            elif presto_val == "total_precipitation":
+                # scaling, and AgERA5 is in mm, Presto expects m
+                values = values / (100 * 1000.0)
+            elif presto_val == "temperature_2m":
+                # remove scaling
+                values = values / 100
+
+            eo_data[:, :, BANDS.index(presto_val)] = values
+
+            mask[:, :, IDX_TO_BAND_GROUPS[presto_val]] = np.clip(
```

Comment:
In these lines we are creating the binary mask that tells Presto which values to ignore (`~idx_valid`).

The `.clip` is just to make sure the mask never has values greater than `1`.

Reply:
In which situation would such values be greater than `1` and would `clip` do exactly this without warning that it doesn't seem a normal situation that these original values were outside the expected range?

Reply:
Yes this is probably excessively defensive - I copied this over from the [original Eval tasks](https://github.com/nasaharvest/presto/blob/main/presto/eval/eurosat_eval.py#L179). 

Originally, this was necessary because the dataloader would return a mask *and* the user can provide their own mask. So if the user wants to mask out a value which the dataloader has already masked, then the `clip` would be necessary. 

However, it can't happen in this case so its safe to remove.

Reply:
Removed